test/test-generate-pras.jl

```diff
@@ -0,0 +1,241 @@
+        PSY.get_name.(
+            PSY.get_components(
+                x -> PSY.get_available(x) && PSY.get_rating(x) > 0,
+                Union{PSY.HydroDispatch, PSY.RenewableGen, PSY.ThermalGen},
+                rts_da_sys,
+            )
+        )
+    storage_names = PSY.get_name.(PSY.get_components(PSY.Storage, rts_da_sys))
+    generatorstorage_names =
+        PSY.get_name.(
+            PSY.get_components(
+                x -> PSY.get_available(x) && PSY.get_rating(x) > 0,
+                Union{PSY.HydroEnergyReservoir, PSY.HybridSystem},
+                rts_da_sys,
+            )
+        )
+    line_names =
+        PSY.get_name.(
+            PSY.get_components(PSY.Branch, rts_da_sys) do c
+                PSY.get_available(c) && !(
+                    c isa PSY.TapTransformer ||
+                    c isa PSY.Transformer2W ||
+                    c isa PSY.PhaseShiftingTransformer
+                )  # From definitions.jl
+                PSY.get_area(PSY.get_from_bus(c)) != PSY.get_area(PSY.get_to_bus(c))
+            end
+        )
+
+    # Make a PRAS System from PSY-4.X System
+    rts_pras_sys = generate_pras_system(rts_da_sys, PSY.Area)
+    @test rts_pras_sys isa PRASInterface.PRAS.SystemModel
+
+    @test test_names_equal(rts_pras_sys.regions.names, area_names)
+    @test test_names_equal(rts_pras_sys.generators.names, generator_names)
+    @test test_names_equal(rts_pras_sys.storages.names, storage_names)
+    @test test_names_equal(rts_pras_sys.generatorstorages.names, generatorstorage_names)
+    @test test_names_equal(rts_pras_sys.lines.names, line_names)
+
+    # Test that timestamps look right
+    # get time series length
+    psy_ts = first(PSY.get_time_series_multiple(rts_da_sys, type=PSY.SingleTimeSeries))
+    @test all(
+        TimeSeries.timestamp(psy_ts.data) .== collect(DateTime.(rts_pras_sys.timestamps)),
+    )
+
+    # Test capacities
+    # 201_HYDRO_4 should have max active power from time series
+    idx = findfirst(x -> x == "201_HYDRO_4", rts_pras_sys.generators.names)
+    hydro_component = PSY.get_component(PSY.HydroDispatch, rts_da_sys, "201_HYDRO_4")
+    max_capacity =
+        floor.(
+            PSY.get_time_series_values(
+                PSY.SingleTimeSeries,
+                hydro_component,
+                "max_active_power",
+            )
+        )
+    @test rts_pras_sys.generators.capacity[idx, 1] == max_capacity[1]
+    @test all(rts_pras_sys.generators.capacity[idx, :] .== max_capacity)
+
+    thermal_component = PSY.get_component(PSY.ThermalStandard, rts_da_sys, "322_CT_6")
+    @test array_all_equal(
+        rts_pras_sys.generators.capacity[
+            findfirst(x -> x == "322_CT_6", rts_pras_sys.generators.names),
+            :,
+        ],
+        Int(floor(PSY.get_max_active_power(thermal_component))),
+    )
+
+    load_values = zeros(Float64, length(rts_pras_sys.regions.names), length(psy_ts))
+    for load in PSY.get_components(PSY.PowerLoad, rts_da_sys)
```

Comment:
Should we filter for available PowerLoads. I think RTS-GMLC built by PSB has PowerLoads but this will be changed to StandardLoad soon because that is out default now.

Reply:
Changed to StaticLoad